local/src/core/YamlParser.cc

```diff
@@ -527,11 +550,30 @@ YamlParser::parse_fields_and_rules(
             } else {
               fields._rules.emplace(name, tester);
             }
-          } else {
-            // No verification directive was specified.
-            if (assume_equality_rule) {
-              fields._rules.emplace(name, RuleCheck::make_equality(name, std::move(values)));
+            continue;
+          }
+
+          // Attempt with "not" instead of "as"
+          auto const rule_type_node_not = ValueNode[YAML_RULE_TYPE_MAP_KEY_NOT];
+          if (rule_type_node_not) {
+            TextView rule_type{rule_type_node_not.Scalar()};
+            std::shared_ptr<RuleCheck> tester =
+                RuleCheck::make_rule_check(name, std::move(values), rule_type, true, is_nocase);
+            if (!tester) {
+              errata.error(
+                  "Field rule at {} does not have a valid directive ({})",
+                  node.Mark(),
+                  rule_type);
+              continue;
+            } else {
+              fields._rules.emplace(name, tester);
             }
+            continue;
+          }
+
+          // No verification directive was specified.
+          if (assume_equality_rule) {
+            fields._rules.emplace(name, RuleCheck::make_equality(name, std::move(values), false, is_nocase));
```

Comment:
Same comment here as concerning `false` above.

Reply:
Reworked to avoid call with false.